asset-transfer-basic/application-gateway-java/src/main/java/application/java/App.java

```diff
@@ -0,0 +1,288 @@
+            // Any error that originates from a peer or orderer node external to the gateway will have its details
+            // embedded within the gRPC status error. The following code shows how to extract that.
+			e.printStackTrace();
+			for (ErrorDetail detail : e.getDetails()) {
+            	System.out.println("\n"+ "Error from endpoint: " + "\n" + "address: " + detail.getAddress() + "\n" +
+                        "mspId: " + detail.getMspId() + "\n" +"message:" + detail.getMessage() + "\n");
+            	System.out.println("Transaction ID: " + e.getTransactionId());
+            }
+			
+        } catch (CommitException e) {
+        	System.out.println("Transaction ID: " + e.getTransactionId() + " failed to commit with status code: " + e.getCode());
+        	e.printStackTrace();
+        } catch (Throwable error) {
+        	System.out.println("******** FAILED to run the application:" + error);
+			error.printStackTrace();
+		
+		} finally {
+			channel.shutdownNow().awaitTermination(5, TimeUnit.SECONDS);
+			
+		}
+	}
+
+	/**
+	 * This type of transaction would typically only be run once by an application the first time it was started after its
+	 * initial deployment. A new version of the chaincode deployed later would likely not need to run an "init" function.
+	 */
+	private static void initLedger(Contract contract) throws GatewayException, CommitException {
+
+		System.out.println("\n" + "--> Submit Transaction: InitLedger, function creates the initial set of assets on the ledger");
+		contract.submitTransaction("InitLedger");
+		System.out.println("*** Transaction committed successfully" + "\n");
+		
+
+	}
+	
+	/**
+	 * Evaluate a transaction to query ledger state.
+	 */
+	private static void getAllAssets(Contract contract) throws GatewayException, CommitException {
+
+		System.out.println("\n" + "--> Evaluate Transaction: GetAllAssets, function returns all the current assets on the ledger");
+		byte[] result = contract.evaluateTransaction("GetAllAssets");
+		System.out.println("*** Result:" + prettyJson(result));
+
+	}
+
+	/**
+	 * Submit a transaction synchronously, blocking until it has been committed to the ledger.
+	 */
+	private static void createAsset(Contract contract) throws GatewayException, CommitException {
+
+		System.out.println("\n" + "--> Submit Transaction: CreateAsset, creates new asset with ID, Color, Size, Owner and AppraisedValue arguments");
+
+		contract.submitTransaction("CreateAsset", assetId, "yellow", "5", "Tom", "1300");
+		
+		System.out.println("*** Transaction committed successfully");
+	}
+
+	/**
+	 * Submit transaction asynchronously, allowing the application to process the smart contract response (e.g. update a UI)
+	 * while waiting for the commit notification.
+	 */
+	private static void transferAssetAsync(Contract contract) throws GatewayException, CommitException {
+
+		// Submit a transaction, blocking until the transaction has been committed on
+		// the ledger.
```

Comment:
This comment is wrong. It isn't blocking until after the commit. I'd suggest removing all these in-line comments within the methods, and just leave method-level comments as in the TypeScript sample.